components/raftstore/src/store/fsm/store.rs

```diff
@@ -234,18 +234,17 @@ pub struct PollContext<T, C: 'static> {
 
 impl<T, C> HandleRaftReadyContext for PollContext<T, C> {
     #[inline]
-    fn kv_wb(&self) -> &WriteBatch {
-        &self.kv_wb
+    fn take_kv_wb(&mut self) -> WriteBatch {
+        self.kv_wb.take().unwrap_or_default()
```

Comment:
`Default` doesn't work with the WriteBatch trait in `engine_traits`, at least as currently designed, since those are produced from factory methods on the engine type.

I think the `or_default` here is an anti-pattern anyway since it discourages precise thinking about invariants - if this option is ever not filled with a WriteBatch then some piece of code has made a mistake with the `take/save` patterns here.

Reply:
Ok. I will use 'new' instead.